### ISSUE TITLE
Type error from `@smui/menu`

### ISSUE THREAD

**Issue by tuckergordon** [hperrin/svelte-material-ui#622]
**Describe the bug**
There is a type error stemming from `@smui/menu`. I actually ran into it while using `@smui/select`. You can see this error in this simple [stackblitz](https://stackblitz.com/edit/sveltejs-kit-template-default-egyygs?description=The%20default%20SvelteKit%20template,%20generated%20with%20create-svelte&file=src%2Froutes%2FSelectExample.svelte,src%2Froutes%2F%2Bpage.svelte&title=SvelteKit%20Default%20Template)

```
/home/projects/sveltejs-kit-template-default-egyygs/node_modules/@smui/menu/src/SelectionGroupIcon.ts:8:3
Error: Types of construct signatures are incompatible.
  Type 'new (options: ComponentConstructorOptions<{ getElement?: (() => HTMLSpanElement) | undefined; use?: ActionArray | undefined; class?: string | undefined; slot?: string | null | undefined; style?: string | ... 1 more ... | undefined; ... 198 more ...; 'on:fullscreenerror'?: EventHandler<...> | ... 1 more ... | undefined; }>) => Graphic__SvelteComponent_' is not assignable to type 'new <Props extends Record<string, any> = any, Events extends Record<string, any> = any, Slots extends Record<string, any> = any>(options: ComponentConstructorOptions<Props>) => SvelteComponent<...>'.
    Construct signature return types 'Graphic__SvelteComponent_' and 'SvelteComponent<Props, Events, Slots>' are incompatible.
      The types of '$$prop_def' are incompatible between these types.
        Type '{ getElement?: (() => HTMLSpanElement) | undefined; use?: ActionArray | undefined; class?: string | undefined; slot?: string | null | undefined; style?: string | ... 1 more ... | undefined; ... 198 more ...; 'on:fullscreenerror'?: EventHandler<...> | ... 1 more ... | undefined; }' is not assignable to type 'Props'.
          '{ getElement?: (() => HTMLSpanElement) | undefined; use?: ActionArray | undefined; class?: string | undefined; slot?: string | null | undefined; style?: string | ... 1 more ... | undefined; ... 198 more ...; 'on:fullscreenerror'?: EventHandler<...> | ... 1 more ... | undefined; }' is assignable to the constraint of type 'Props', but 'Props' could be instantiated with a different subtype of constraint 'Record<string, any>'. 
  class: 'mdc-menu__selection-group-icon',
  component: Graphic,
});
```

**To Reproduce**
1. `npm i -D @smui/select`
2. Import `@smui/select` into a component
3. `npm run check` to run `svelte-check` 

**Additional context**
Looks like this showed up in #610 but didn't get resolved as part of #609

**Comment by locus2k** [comment]
I too am getting the same issue

**Comment by vhscom** [comment]
To reproduce create a svelte skeleton project with JS or TS type checking:
```shell
npm create svelte@latest smui-issue-662-type-error
```

Install dependencies:
```shell
npm i -D @smui/menu
```

Add the following to `+page.svelte` and save the file:
```svelte
<script>
  import Menu from "@smui/menu";
</script>
```

Then run `npm run check` and observe the error: https://app.warp.dev/block/lAQifBvle8LjdX66mkbhCD

**Comment by vhscom** [comment]
Pinned and backed down to several other beta versions 14, 13, 12 and 11, then down to `beta.0` and all throw the same error. Given this error did not exist in `beta.0` and `@sveltejs/kit@1.0.1` it seems this is some kind of corner case. Given it's common to run `svelte-check` as a CI trigger I hope to see a workaround for this if the fix isn't immediately obvious.

**Comment by vhscom** [comment]
@tuckergordon Have you found a workaround for this?

**Comment by vhscom** [comment]
Possibly related to https://github.com/sveltejs/language-tools/issues/1100

**Comment by vhscom** [comment]
Possible root cause: https://github.com/sveltejs/language-tools/issues/1056

**Comment by cabreraalex** [comment]
Any way round this?

**Comment by ihr-it-projekt** [comment]
Same issue, any solution in pipeline?

How i can suppress  such an error? To make my bild pipeline working?

**Comment by ihr-it-projekt** [comment]
@hperrin
Any plans to fix that? 
I have tried, but i fail to understand the problem... sorry.

**Comment by hperrin** [comment]
This is fixed in 7.0.0-beta.16!
